Azure/azure-cli
# [Compute] Fix #19599: `az vm create`: Fix the issue that `--nic-delete-option` not working when no `--nics` is provided.

Issue: #19599 

**Description**

Fix #19599 nic-delete-option was not working when no nics are provided with it. Also throw exception when the user specifies individual delete options and nics are not provided. 
**Testing Guide**

`az vm create --nic-delete-option Delete --resource-group {rg} --name {vm-name} --image UbuntuLTS

---

This checklist is used to make sure that common guidelines for a pull request are followed.
